src/main/java/net/syscon/elite/service/BookingService.java

```diff
@@ -388,6 +402,53 @@ private void validateActivity(final Long activityId) {
         return bookingRepository.getBookingVisits(bookingId, fromDate, toDate, sortFields, sortOrder);
     }
 
+    @VerifyBookingAccess
+    public Page<VisitWithVisitors<Visit>> getBookingVisitsWithVisitor(final @NotNull Long bookingId, final LocalDate fromDate, final LocalDate toDate, final String visitType, final Pageable pageable) {
+        final var visits = visitRepository.findAllByBookingId(bookingId, pageable);
+
+        final var visitsWithVisitors = visits.stream()
+                .filter(visit -> fromDate == null || visit.getStartTime().isAfter(fromDate.atStartOfDay()))
+                .filter(visit -> toDate == null || visit.getStartTime().isBefore(toDate.atTime(LocalTime.of(23, 59))))
```

Comment:
Could maybe use before LocalTime.MAX here so you're not caught out by start times of 23:59 exactly?

Reply:
Learn something new every day!